tests/ui/unnecessary_min.rs

```diff
@@ -0,0 +1,45 @@
+#![allow(unused)]
+#![warn(clippy::unnecessary_min)]
+
+fn main() {
+    const A: i64 = 45;
+    const B: i64 = -1;
+    const C: i64 = const_fn(B);
+    let _ = (A * B).min(B); // Both are constants
+    let _ = C.min(B); // Both are constants
+    let _ = B.min(C); // Both are constants
```

Comment:
I don't know whether clippy has a policy for these sorts of things, but linting using the values of constants makes me wary.  It would be possible to change `A`, `B`, or `C` in such a way that the use of `min` or `max` do actually do something, especially if it's something like `min(CONST_LIMIT, user_input)`.

Reply:
>I don't know whether clippy has a policy for these sorts of things, but linting using the values of constants makes me wary. It would be possible to change A, B, or C in such a way that the use of min or max do actually do something

I don't think i get your point.
In your commented block of code all "variables" are constants so afaik it is not possible that one of them can change.
In the case of
>`min(CONST_LIMIT, user_input)`

there should be only a lint emitted if `CONST_LIMIT` is either the minimum or maximum value. If it is the minimum value, then the result will always be the minumum regardless of the `user_input` and if it is the maximum value then the result will always be the `user_input`.

Or did you mean something like that?:
```rust
let mut m = u32::MAX;
for _ in 0..1000{
   m = m.min(user_input);
}
```
In this situation nothing should be linted.
I'll add a test to confirm it.

thank you for your feedback. I appreciate it.